### PR TITLE
Invoke-DbaQuery - add cross-platform support updating URI code

### LINKED ISSUE #6145: [Bug] Invoke-DbaQuery -File fails on Linux
#### PowerShell version:
```powershell
Name                           Value
----                           -----
PSVersion                      6.2.3
PSEdition                      Core
GitCommitId                    6.2.3
OS                             Linux 3.10.0-1062.1.2.el7.x86_64 #1 SMP Mon Sep …
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0



#### dbatools Module version:

Name    : dbatools
Path    : /home/josh/.local/share/powershell/Modules/dbatools/1.0.58/dbatools.psd1
Version : 1.0.58
```

```powershell
LCID             Name             DisplayName
----             ----             -----------
1033             en-US            English (United States)
```

#### SQL Server: 

```sql
Column1
-------
Microsoft SQL Server 2017 (RTM-CU17) (KB4515579) - 14.0.3238.1 (X64) …
```

```sql
Column1
-------
us_english
```

#### Host used

- [ ] powershell.exe
- [ ] ISE
- [ ] VS Code
- [x] Other (please specify) pwsh core

If anything other than powershell.exe was used, please confirm that you can duplicate the issue with powershell.exe

- [ ] Still buggy in powershell.exe

#### Errors Received

```powershell
writeErrorStream      : True
PSMessageDetails      :
Exception             : System.Exception: Could not resolve path
                        /home/josh/test.sql as filesystem object
TargetObject          :
CategoryInfo          : NotSpecified: (:) [Write-Error], Exception
FullyQualifiedErrorId : dbatools_Invoke-DbaQuery,Stop-Function
ErrorDetails          : Could not resolve path /home/josh/test.sql as
                        filesystem object
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at Stop-Function, /home/josh/.local/share/powershell/Mo
                        dules/dbatools/1.0.58/allcommands.ps1: line 81516
                        at Invoke-DbaQuery<Begin>, /home/josh/.local/share/powe
                        rshell/Modules/dbatools/1.0.58/allcommands.ps1: line
                        46352
                        at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo : {0, 1}
```

#### Steps to Reproduce

1. Create a .sql file on a unix host.
2. Execute a `Invoke-DbaQuery` with `-File`

#### Expected Behavior

Expected the `Invoke-DbaQuery` to execute the SQL `-file`.

#### Actual Behavior

image

Believe the ultimate issue is with lines 213 to 221 of `Invoke-DbaQuery` where it is looking for Scheme type of 'file' from the URI, which doesn't seem to be supported on unix systems. example output of line 215 `[uri]$path.fullname` 

```powershell
AbsolutePath   :
AbsoluteUri    :
LocalPath      :
Authority      :
HostNameType   :
IsDefaultPort  :
IsFile         :
IsLoopback     :
PathAndQuery   :
Segments       :
IsUnc          :
Host           :
Port           :
Query          :
Fragment       :
Scheme         :
OriginalString : /home/josh/test.sql
DnsSafeHost    :
IdnHost        :
IsAbsoluteUri  : False
UserEscaped    : False
UserInfo       :
```
Example of the same file on a Windows system in both 5.1 and 6.0
```powershell
PS C:\Users\joshu\Documents> [uri]$path.FullName                                                                        

AbsolutePath   : C:/Users/joshu/Documents/test.sql
AbsoluteUri    : file:///C:/Users/joshu/Documents/test.sql
LocalPath      : C:\Users\joshu\Documents\test.sql
Authority      :
HostNameType   : Basic
IsDefaultPort  : True
IsFile         : True
IsLoopback     : True
PathAndQuery   : C:/Users/joshu/Documents/test.sql
Segments       : {/, C:/, Users/, joshu/...}
IsUnc          : False
Host           :
Port           : -1
Query          :
Fragment       :
Scheme         : file
OriginalString : C:\Users\joshu\Documents\test.sql
DnsSafeHost    :
IdnHost        :
IsAbsoluteUri  : True
UserEscaped    : False
UserInfo       :
```

### PR DESCRIPTION
**Pull request by wsmelton**
## Type of Change

 - [x] Bug fix (non-breaking change, fixes #6145 )
 - [ ] New feature (non-breaking change, adds functionality, fixes # )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1)
 - [ ] Adding code coverage to existing functionality
 - [ ] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/sqlcollaborative/appveyor-lab ?
 - [ ] Nunit test is included
 - [ ] Documentation
 - [ ] Build system
 

### Purpose
Utilize the `uri::new()` constructor over just trying to declare a variable as a URI type. This ensures the proper object is created so the validation can be performed properly.

Added an internal command for testing version of PowerShell and making it a bit cleaner in code. Included examples to show usage. So we will be able to do things like `Test-PsVersion -Minimum 6.2` or something of that nature if we want to start including newer features (planning for PS 7 ;) )

### CODE REVIEW

File: functions/Invoke-DbaQuery.ps1 [hunk 1]
```diff
@@ -182,7 +182,7 @@ function Invoke-DbaQuery {
                         $files += $item.FullName
                     }
                     "System.String" {
-                        $uri = [uri]$item
+                        $uri = [uri]::New($item)
```
Comment (potatoqualitee) [inline functions/Invoke-DbaQuery.ps1]: ::New breaks in PS 3 😢

Comment (wsmelton) [reply functions/Invoke-DbaQuery.ps1]: What version of .NET are you running?

**Comment by potatoqualitee** [comment]
the minimal required to run PS 3. this was an issue around 1.0 that had to be undone across the board and @niphlod has it on his todo to make it fail in tests cuz we kept forgetting. i finally remembered 🎉 !

**Comment by wsmelton** [comment]
Well there is no workaround for this to make it work in PS3 and PS Core. I can change it where this code will only get used in PS4 and above, then in PS3 they just can't use files.

**Comment by potatoqualitee** [comment]
could they use files before? i like the idea of handling it diff ways 💯

**Comment by wsmelton** [comment]
Yes, the old way worked just not on PS 6+. As well, it was just not the proper way of doing it.

**Comment by potatoqualitee** [comment]
k please keep the functionality for ps 3 then handle core in an else 👍

**Comment by joshcorr** [comment]
@wsmelton Do you need any help with this PR? I am wanting to work on some CrossPlat support that relies on this fix.

**Comment by wsmelton** [comment]
It is ready to be merged.

We should not be using this command in other functions. We use the query() method to execute SQL code.

**Comment by wsmelton** [comment]
Failures are not related to this PR.

**Pull request merged**